src/index.ts

```diff
@@ -1,117 +1,47 @@
 import { RollupContext } from "./rollupcontext";
-import { IContext, ConsoleContext, IRollupContext, VerbosityLevel } from "./context";
+import { ConsoleContext, IRollupContext, VerbosityLevel } from "./context";
 import { LanguageServiceHost } from "./host";
-import { TsCache, convertDiagnostic, ICode, IDiagnostics } from "./tscache";
-import * as ts from "typescript";
-import * as fs from "fs-extra";
-import * as path from "path";
-import * as _ from "lodash";
-import * as colors from "colors/safe";
+import { TsCache, convertDiagnostic, ICode } from "./tscache";
+import { createLanguageService, version, createDocumentRegistry, OutputFile, ParsedCommandLine, sys, LanguageService, nodeModuleNameResolver } from "typescript";
 import * as resolve from "resolve";
+import {defaults, endsWith, concat, find, isFunction, get, each} from "lodash";
+import { IRollupOptions } from "./irollup-options";
+import { IOptions } from "./ioptions";
+import { Partial } from "./partial";
+import { parseTsConfig } from "./parse-ts-config";
+import { printDiagnostics } from "./print-diagnostics";
+import { TSLIB, tslibSource } from "./tslib";
+import {blue, red, yellow} from "colors/safe";
+import {join, relative, dirname} from "path";
 
 // tslint:disable-next-line:no-var-requires
 const createFilter = require("rollup-pluginutils").createFilter;
-
-function getOptionsOverrides(): ts.CompilerOptions
-{
-	return {
-		module: ts.ModuleKind.ES2015,
-		noEmitHelpers: true,
-		importHelpers: true,
-		noResolve: false,
-	};
-}
-
-// The injected id for helpers.
-const TSLIB = "tslib";
-let tslibSource: string;
-try
-{
-	// tslint:disable-next-line:no-string-literal no-var-requires
-	const tslibPath = require.resolve("tslib/" + require("tslib/package.json")["module"]);
-	tslibSource = fs.readFileSync(tslibPath, "utf8");
-} catch (e)
+let watchMode = false;
```

Comment:
It seems to me what you are implying is that variables defined in the global scope is accessible from outside the module they are declared within. That is incorrect. For it to persist, it must be bound to the global object (`global` in Node) and execute within the same execution environment.

So, running multiple instances would not make those variables interfere with one another.